server/src/test/kotlin/com/kamelia/jellyfish/rest/AuthTest.kt

```diff
@@ -1,48 +1,282 @@
 package com.kamelia.jellyfish.rest
 
 import com.kamelia.jellyfish.client
+import com.kamelia.jellyfish.core.TokenData
 import com.kamelia.jellyfish.login
+import com.kamelia.jellyfish.loginBlocking
 import io.ktor.client.request.bearerAuth
+import io.ktor.client.request.delete
+import io.ktor.client.request.get
 import io.ktor.client.request.patch
+import io.ktor.client.statement.bodyAsText
 import io.ktor.http.HttpStatusCode
+import io.ktor.server.config.ApplicationConfig
 import io.ktor.server.testing.testApplication
+import kotlinx.coroutines.delay
+import kotlinx.serialization.json.Json
 import org.junit.jupiter.api.DisplayName
 import org.junit.jupiter.api.MethodOrderer
-import org.junit.jupiter.api.Order
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
 import org.junit.jupiter.api.TestMethodOrder
 import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
 
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @TestMethodOrder(MethodOrderer.OrderAnnotation::class)
 class AuthTest {
 
-    @DisplayName("Logging in with correct credentials")
+    @DisplayName("Logging in with incorrect username")
     @Test
-    @Order(1)
-    fun loggingIn() = testApplication {
-        val (status, _) = login("user1", "password")
-        assertEquals(HttpStatusCode.OK, status)
+    fun loggingInIncorrectUsername() = testApplication {
+        val (status, _) = login("wrongUser", "password")
+        assertEquals(HttpStatusCode.Unauthorized, status)
     }
 
-    @DisplayName("Logging in with incorrect credentials")
+    @DisplayName("Logging in with incorrect password")
     @Test
-    @Order(2)
-    fun loggingInIncorrect() = testApplication {
-        val (status, _) = login("user1", "wrong")
+    fun loggingInIncorrectPassword() = testApplication {
+        val (status, _) = login("user1", "wrongPassword")
         assertEquals(HttpStatusCode.Unauthorized, status)
     }
 
-    @DisplayName("Refreshing token")
+    @DisplayName("Performing protected request with valid access token")
     @Test
-    @Order(3)
-    fun refreshToken() = testApplication {
+    fun useValidAccessToken() = testApplication {
+        environment {
+            config = ApplicationConfig("application-auth-test.conf")
+        }
```

Comment:
extract that in a function authTestApplication (or something like that) ; or at least extract the filename in a constant

Reply:
Makes sense